mission_control/navigator_alarm/nodes/event_listeners/odom_kill_listener.py

```diff
@@ -1,52 +1,79 @@
 #!/usr/bin/env python
 import rospy
 from std_msgs.msg import String
+from std_srvs.srv import Trigger, TriggerResponse
+from nav_msgs.msg import Odometry
 from navigator_alarm import single_alarm
 from twisted.internet import reactor
 import time
-import nav_msgs.msg import Odometry
 
 
 class OdomKillListener(object):
     '''
     Will kill the boat when it stops hearing odometry messages or when there is a large
-    discontinuity in state estimation. Only meant to run on the boad as a safety measure.
+    discontinuity in state estimation. Only meant to run on the boat as a safety measure.
     '''
     def __init__(self, timeout=0.5):
         self.timeout = rospy.Duration(timeout)
         self.last_time = rospy.Time.now()
         self.last_position = None
         self.max_jump = 1.0
         self.odom_discontinuity = False
-        self.sub = rospy.Subscriber('/odom', String, self.got_network_msg, queue_size=1)
+        self.killed = False
+        self.odom_listener = rospy.Subscriber('/odom', Odometry, self.got_odom_msg, queue_size=1)
+        self.clear_kill_srv = rospy.Service('clear_odom_kill', Trigger, self.clear_kill)
```

Comment:
I would like this to be an alarm listener instance so that you can clear the alarm from the command line